### ISSUE TITLE
Are the weights used in topological sort?

### ISSUE THREAD

**Issue by david-siqi-liu** [shrimai/Topological-Sort-for-Sentence-Ordering#6]
Correct me if I'm wrong but I don't think weights/logits are used in predicting the order in topological sort.

Personally, I don't know if there is a polynomial-time algorithm to compute the "maximum sum of weights" among all possible topologically sorted paths. Do you know if there is any? I would imagine if we take the computed logits into account, our results would be more accurate.

**Comment by shrimai** [comment]
This is correct. In the results published in the paper, we have not used the weights/logits to predict the order.
I did not explore the idea of using logits to determine the best possible 'topologically sorted path'. In the paper we just pick one of the many possible topologically sorted orders and use them to evaluate our model. Yes, if you take the logits into account then the results should improve. Unfortunately, I cannot help you much with that direction.
You can also use the logits to break the cycles (break the edge with lowest weight). Right now, I am simply breaking the last edge at which I detect a cycle.
